apps/webapp/src/components/swap/swap-form.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button } from '@penumbra-zone/ui';
+import { useToast } from '@penumbra-zone/ui/components/ui/use-toast';
+import InputToken from '../shared/input-token.tsx';
+import { useLoaderData } from 'react-router-dom';
+import { AccountBalance } from '../../fetchers/balances.ts';
+import { useStore } from '../../state';
+import { swapSelector } from '../../state/swap.ts';
+import { AssetOutBox } from './asset-out-box.tsx';
+
+export const SwapForm = () => {
+  const accountBalances = useLoaderData() as AccountBalance[];
+  const { toast } = useToast();
+  const { assetIn, setAssetIn, amount, setAmount, initiateSwapTx } = useStore(swapSelector);
+
+  return (
+    <form
+      className='flex flex-col gap-4 xl:gap-3'
+      onSubmit={e => {
+        e.preventDefault();
+        void initiateSwapTx(toast);
+      }}
+    >
+      <InputToken
+        label='Amount to swap'
+        placeholder='Enter an amount'
+        className='mb-1'
+        selection={assetIn}
+        setSelection={setAssetIn}
+        value={amount}
+        onChange={e => {
+          if (Number(e.target.value) < 0) return;
+          setAmount(e.target.value);
```

Comment:
See my note in the Zustand state — shouldn't this be converted to a number first?